### PR TITLE
Use Loggers.getLogger in TransportShardBulkAction

### PR DESCRIPTION
**Pull request by khushbr**
### Description
Use Loggers.getLogger in TransportShardBulkAction

### Check List
~- [ ] New functionality includes testing.~
  - [x] All tests pass
~- [ ] New functionality has been documented.~
~- [ ] New functionality has javadoc added~
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
~- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))~
~- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by kkmr — APPROVED** [review]

**Review by mgodwan — APPROVED** [review]

**Review by shwetathareja — APPROVED** [review]

**Comment by andrross** [comment]
Test failures all seem to stem from:

```
        Caused by:
        java.lang.IllegalArgumentException: if you don't need a prefix then use a regular logger
            at org.opensearch.common.logging.PrefixLogger.<init>(PrefixLogger.java:95)
            at org.opensearch.common.logging.Loggers.getLogger(Loggers.java:89)
            at org.opensearch.action.bulk.TransportShardBulkAction.<clinit>(TransportShardBulkAction.java:129)
            ... 88 more
```

File: server/src/main/java/org/opensearch/action/bulk/TransportShardBulkAction.java [hunk 1]
```diff
@@ -126,7 +126,7 @@ public class TransportShardBulkAction extends TransportWriteAction<BulkShardRequ
 
     public static final String ACTION_NAME = BulkAction.NAME + "[s]";
 
-    private static final Logger logger = LogManager.getLogger(TransportShardBulkAction.class);
+    private static final Logger logger = Loggers.getLogger(TransportShardBulkAction.class);
```
Comment (andrross) [inline server/src/main/java/org/opensearch/action/bulk/TransportShardBulkAction.java:129]: What's the reason for this change? The `Logger getLogger(Class<?> clazz, String... prefixes)` method seems to explicitly disallow this usage.

Side note, seems like it would be better to make that method signature `Logger getLogger(Class<?> clazz, String prefix, String... prefixes)` so you can enforce that constraint at compile time versus runtime.

Comment (khushbr) [reply server/src/main/java/org/opensearch/action/bulk/TransportShardBulkAction.java:129]: @andrross The change was made with intent to surface underlying Class failure when applying Bulk operation, via `PrefixLogger`

I will close this PR for now; this requires more re-work.
